### PR TITLE
feat(inputs.opcua_listener): Allow selecting the deadband mode 'None'

### LINKED ISSUE #18464: [inputs.opcua_listener] Support deadband type of 'None' for data_change_filter
### Use Case

If you want the subscription to output a metric even when the value hasn't changed, you need to subscribe with the data_change_filter trigger set to StatusValueTimestamp, but you may not want (and the node might not support) deadband filtering. The data_change_filter should support the deadband type of 'None' in this case. See the relevant part of the spec that lists this option: https://reference.opcfoundation.org/Core/Part4/v104/docs/7.17.2

### Expected behavior

Ability to subscribe to an opc-ua node and get value updates every `sampling_interval` even when the value hasn't changed.

### Actual behavior

Config doesn't validate when using the deadband type of None, and if you try to use Absolute, or Percent deadband type the opc-ua server can reject it with a `BadFilterNotAllowed` error depending on node type.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by gazpachoking**
## Summary

Some opc-ua nodes do not support using a deadband filter, but you might still want to specify that they should use a different data_change_filter trigger type, such as StatusValueTimestamp. Allowing the 'None' option for the deadband type permits this.

## Checklist

- [x] No AI generated code was used in this PR
- [ ] AI generated code used in this PR follows the [InfluxData Policy on AI-Generated Code Contributions][policy]

[policy]: https://www.influxdata.com/ai-generated-code-contributions-policy

## Related issues

resolves #18464

### CODE REVIEW

File: plugins/inputs/opcua_listener/subscribe_client.go
Comment (skartikey) [inline plugins/inputs/opcua_listener/subscribe_client.go]: The validation requires `deadband_value` to be set and non-negative regardless of deadband type. When `deadband_type = "None"`, the deadband value is meaningless, users shouldn't be
  forced to set it. The validation should skip the value checks for "None":

```
  case params.DeadbandType != input.None && params.DeadbandValue == nil:
      return errors.New("deadband_value was not set")
  case params.DeadbandValue != nil && *params.DeadbandValue < 0:
      return errors.New("negative deadband_value not supported")
```

  And line 77 would need a nil guard or default to 0:
  DeadbandValue: 
```
func() float64 {
      if monParams.DataChangeFilter.DeadbandValue != nil {
          return *monParams.DataChangeFilter.DeadbandValue
      }
      return 0
  }(),
```

  Without this, the whole point of the feature is undermined; users who don't want a deadband filter must still configure a dummy value.

**Review by skartikey — COMMENTED** [review]
@gazpachoking Thanks for the PR!

There are no test changes. At minimum, the PR should:
  - Add a test case in checkDataChangeFilterParameters validation for deadband_type = "None" being accepted
  - Add a test for deadband_type = "None" without deadband_value (which currently fails but should succeed per other comment)

**Comment by gazpachoking** [comment]
Good points, thanks! I added a test, as well as made deadband value optional when deadband type is none.

File: plugins/inputs/opcua_listener/subscribe_client.go [hunk 1]
```diff
@@ -72,9 +73,14 @@ func assignConfigValuesToRequest(req *ua.MonitoredItemCreateRequest, monParams *
...
-				DeadbandType:  uint32(ua.DeadbandTypeFromString(string(monParams.DataChangeFilter.DeadbandType))),
-				DeadbandValue: *monParams.DataChangeFilter.DeadbandValue,
+				Trigger:      ua.DataChangeTriggerFromString(string(monParams.DataChangeFilter.Trigger)),
+				DeadbandType: uint32(ua.DeadbandTypeFromString(string(monParams.DataChangeFilter.DeadbandType))),
+				DeadbandValue: func() float64 {
+					if monParams.DataChangeFilter.DeadbandValue != nil {
+						return *monParams.DataChangeFilter.DeadbandValue
+					}
+					return 0
+				}(),
```
Comment (skartikey) [inline plugins/inputs/opcua_listener/subscribe_client.go]: This is unnecessarily clever(IIFE for nil-safe dereference). A simpler approach would be to declare a local variable before the struct literal, or restructure the code to set DeadbandValue conditionally. For example:                        
   
```
  var deadbandValue float64                                                                                                                                                                                        
  if monParams.DataChangeFilter.DeadbandValue != nil {                                                                                                                                                             
      deadbandValue = *monParams.DataChangeFilter.DeadbandValue                                                                                                                                                    
  }                                                                                                                                                                                                                
                                                                                                                                                                                                                   
  req.RequestedParameters.Filter = ua.NewExtensionObject(                                                                                                                                                          
      &ua.DataChangeFilter{                                                                                                                                                                                        
          Trigger:       ua.DataChangeTriggerFromString(string(monParams.DataChangeFilter.Trigger)),                                                                                                               
          DeadbandType:  uint32(ua.DeadbandTypeFromString(string(monParams.DataChangeFilter.DeadbandType))),                                                                                                       
          DeadbandValue: deadbandValue,                                                                                                                                                                            
      },                                                                                                                                                                                                           
  ) 
```

Comment (gazpachoking) [reply plugins/inputs/opcua_listener/subscribe_client.go]: Lol, I did it that way because that's what you had in your last comment 😆 I'll change it up tonight

Comment (skartikey) [reply plugins/inputs/opcua_listener/subscribe_client.go]: @gazpachoking Ha, you're absolutely right, that's on me! I suggested the IIFE in my first comment without thinking it through, then flagged it on the second pass. Sorry for the back and forth. The local variable approach is the cleaner one, so let's go with that. Thanks for being a good sport about it!
